src/generated/xproto.rs

```diff
@@ -6489,273 +7970,292 @@ impl From<Gravity> for Option<u32> {
         Some(u32::from(input))
     }
 }
-
-/// Opcode for the CreateWindow request
-pub const CREATE_WINDOW_REQUEST: u8 = 1;
-/// Auxiliary and optional information for the create_window function.
-#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
-pub struct CreateWindowAux {
-    pub background_pixmap: Option<Pixmap>,
-    pub background_pixel: Option<u32>,
-    pub border_pixmap: Option<Pixmap>,
-    pub border_pixel: Option<u32>,
-    pub bit_gravity: Option<u32>,
-    pub win_gravity: Option<u32>,
-    pub backing_store: Option<u32>,
-    pub backing_planes: Option<u32>,
-    pub backing_pixel: Option<u32>,
-    pub override_redirect: Option<Bool32>,
-    pub save_under: Option<Bool32>,
-    pub event_mask: Option<u32>,
-    pub do_not_propogate_mask: Option<u32>,
-    pub colormap: Option<Colormap>,
-    pub cursor: Option<Cursor>,
-}
+
+/// Opcode for the CreateWindow request
+pub const CREATE_WINDOW_REQUEST: u8 = 1;
+/// Auxiliary and optional information for the `create_window` function
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
+pub struct CreateWindowAux {
+    pub background_pixmap: Option<Pixmap>,
+    pub background_pixel: Option<u32>,
+    pub border_pixmap: Option<Pixmap>,
+    pub border_pixel: Option<u32>,
+    pub bit_gravity: Option<u32>,
+    pub win_gravity: Option<u32>,
+    pub backing_store: Option<BackingStore>,
+    pub backing_planes: Option<u32>,
+    pub backing_pixel: Option<u32>,
+    pub override_redirect: Option<Bool32>,
+    pub save_under: Option<Bool32>,
+    pub event_mask: Option<u32>,
+    pub do_not_propogate_mask: Option<u32>,
+    pub colormap: Option<Colormap>,
+    pub cursor: Option<Cursor>,
+}
+impl Serialize for CreateWindowAux {
+    type Bytes = Vec<u8>;
+    fn serialize(&self) -> Self::Bytes {
+        let mut result = Vec::new();
+        self.serialize_into(&mut result);
+        result
+    }
+    fn serialize_into(&self, bytes: &mut Vec<u8>) {
+        if let Some(background_pixmap) = self.background_pixmap {
+            bytes.reserve(4);
+            background_pixmap.serialize_into(bytes);
+        }
+        if let Some(background_pixel) = self.background_pixel {
+            bytes.reserve(4);
+            background_pixel.serialize_into(bytes);
+        }
+        if let Some(border_pixmap) = self.border_pixmap {
+            bytes.reserve(4);
```

Comment:
Why the new `bytes.reserve(4);` line? Isn't this the job of the `Serialize` implementation that is called next? (And for a single element, it makes no sense, since the `Vec` gets one call to `extend_from_slice` and at that point it already knows how much space to allocate and can `reserve` internally)

Reply:
Indeed, the `reserve` is useless if it is followed by a single fixed-size element

Reply:
Fixed (at least in most places)